src/coreclr/jit/lclvars.cpp

```diff
@@ -2051,6 +2051,12 @@ unsigned Compiler::compMap2ILvarNum(unsigned varNum) const
         return (unsigned)ICorDebugInfo::RETBUF_ILNUM;
     }
 
+    if (info.compMethodInfo->args.hasImplicitThis() && varNum == info.compThisArg)
+    {
+        // implicit this doesn't map to any IL arg
```

Comment:
```suggestion
        // implicit this doesn't map to any method signature arg
```
assuming this change is otherwise correct that I am not sure about

Reply:
I reverted those changes since `compMap2ILvarNum` is also used in debug info generation and I don't want to break it accidentally. I put the workaround directly to my function.